src/dmd/initsem.d

```diff
@@ -585,17 +585,25 @@ extern(C++) Initializer initializerSemantic(Initializer init, Scope* sc, ref Typ
 
         auto dil = ci.initializerList[];
 
+        /* If `{ expression }` return the expression initializer
+         */
+        ExpInitializer isBraceExpression()
+        {
+            ExpInitializer ei;
+            return (dil.length == 1 &&
+                    !dil[0].designatorList &&
+                    (ei = dil[0].initializer.isExpInitializer()) !is null)
+                ? ei : null;
```

Comment:
this is much harder to read than
```d
if (dil.length == 1 && !dil[0].designatorList)
    return dil[0].initializer.isExpInitializer();
else 
    return null;
```

Reply:
thanks for the idea, was able to improve it